### ISSUE TITLE
Ballot submission from adjudicators not keeping names

### ISSUE THREAD

**Issue by 0zlw** [TabbycatDebate/tabbycat#1451]
We've run into an issue where ballot submissions being submitted directly from adjudicators are reverting to "Speaker 1/Speaker 2/Speaker 3" when we go to confirm them.

This issue happens where, when adding teams, rather than replacing Speaker 1/Speaker 2/Speaker 3 with speaker names, we add the speaker names instead. We do this as because we are running a schools tournament, we want to keep the Speaker 1/Speaker 2/Speaker 3 options available in the event that schools sub in speakers they have not already registered.

This issue is happening for all submitted marksheets for two of our tournaments running v2.3.2

For example this is a screenshot of what Tabbycat shows:

![image](https://user-images.githubusercontent.com/3196468/81533518-79823d00-93ba-11ea-8b67-d73f6fbf559b.png)

And this is the written marksheet we received from the adjudicator:

#12 - Affirming Junior Olive (235) lost to negating Junior Green (241).
Aff:
1. Amira Stephenson, 68
2. Max Bulley-Brown, 67
3. Alisa Rao, 67
R. Amira Stephenson, 33
Neg:
1. Ryan Antia, 71
2. Alyna Mithani, 68
3. Zahlia Prasad, 69
R. Alyna Mithani, 33

This is what one of the team's speaker names looks like. 

![image](https://user-images.githubusercontent.com/3196468/81533981-47250f80-93bb-11ea-8697-09f65371e3bd.png)

**Comment by tienne-B** [comment]
Are the correct speakers shown in the database view (under Results -> Speaker Scores)?

**Comment by 0zlw** [comment]
Yes! We have confirmed the other marksheet, but this is a marksheet from a debate with the same problem.

![image](https://user-images.githubusercontent.com/3196468/81537682-aede5900-93c1-11ea-97ea-99b57af672d9.png)

**Comment by philipbelesky** [comment]
Hey, thanks for this report. If you’re ok with us looking at your data, DM our Facebook page and I’ll let you know an email address to add as a collaborator on Heroku. I suspect something is making the JavaScript that replaces the names unhappy.

**Comment by tienne-B** [comment]
Okay. This narrows down the problem. The problem seems to be in filling the ballot from the draft result, not from saving the result (although saving it without correcting the faulty initial will override the correct info).

Can't immediately reproduce the problem, but the search is narrow and I'm looking :)

**Comment by 0zlw** [comment]
Have sent a message, otherwise, you can email us directly at online@debating.org.nz.

**Comment by 0zlw** [comment]
Just updating this with some additional information from some stuff I have setup for R1 of a new tournament running v2.3.2

I confirmed this marksheet with the correct speaker names as per the below screenshots.

![image](https://user-images.githubusercontent.com/3196468/83503252-e5b61380-a516-11ea-96a7-84c233e6c3b9.png)

![image](https://user-images.githubusercontent.com/3196468/83503373-139b5800-a517-11ea-8325-84ddf8aaf91c.png)


When I reopen that marksheet despite being confirmed with the correct speaker names, it shows the following.

![image](https://user-images.githubusercontent.com/3196468/83503197-d5059d80-a516-11ea-92e1-02d450ec4935.png)

We kept the original Tabbycat Speaker 1/Speaker 2/Speaker 3. My read of the issue is that, if you don't change the speaker names, i.e. they are Speaker 1/Speaker 2/Speaker 3 as per the screenshot below, Tabbycat will by default populate them with Speaker 1/Speaker 2/Speaker 3.

![image](https://user-images.githubusercontent.com/3196468/83503532-4e04f500-a517-11ea-97a0-9a804faaa9e0.png)

In this case because we have kept the default speakers, but also added some Tabbycat overwrites the entered speaker names with the defaults which is causing the issue.

This is what happened when I deleted Speaker 1/Speaker 2/Speaker 3 from the database area.

![image](https://user-images.githubusercontent.com/3196468/83503646-83a9de00-a517-11ea-896c-79544adc9e6a.png)

This is then what happened when I manually added Speaker 1/Speaker 2/Speaker 3 back.

![image](https://user-images.githubusercontent.com/3196468/83503726-9fad7f80-a517-11ea-97d0-89520fb9befb.png)

For comparison sake:

![image](https://user-images.githubusercontent.com/3196468/83503743-a89e5100-a517-11ea-8443-0c7db238019e.png)

![image](https://user-images.githubusercontent.com/3196468/83503761-b227b900-a517-11ea-874f-a6e3de06692b.png)

**Comment by tienne-B** [comment]
Very strange problem, as Tabbycat doesn't put defaults for speakers, and the default "Speaker #" speakers are no different to named speakers. If it's not @philipbelesky's theory, then we get to farfetched ideas like problems with the admin interface or caching. Philip, did you find anything?

**Comment by philipbelesky** [comment]
I haven't been able to reproduce this myself, although the notes above pretty much confirm what I was thinking: that its a problem with the javascript on that form page. Or to be precise, [this block in `enter_results.js`](https://github.com/TabbycatDebate/tabbycat/blob/develop/tabbycat/templates/js-bundles/enter_results.js#L5). 

I think this works out so that if you use generic speaker names (exactly "Speaker 1") then it essentially automatically selects the values in order. This may be a holdover from something like the WADL league options where you only care about team wins; not speakers.

What this means is that if you use, say, "Speaker #1", as your placeholder value it should no longer do this. Could you confirm that @0zlw? I also wouldn't be averse to removing this functionality, or perhaps making it only apply on a less commonly used placeholder value.
